src/main/java/com/amazon/dlic/auth/http/saml/HTTPMetadataResolver.java

```diff
@@ -0,0 +1,156 @@
+ * compatible open source license.
+ */
+
+package com.amazon.dlic.auth.http.saml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Timer;
+
+import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.protocol.HttpClientContext;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.http.HttpStatus;
+
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+import org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HTTPMetadataResolver extends AbstractReloadingMetadataResolver {
+    private final Logger log = LoggerFactory.getLogger(HTTPMetadataResolver.class);
+    private HttpClient httpClient;
+    private URI metadataURI;
+    private String cachedMetadataETag;
+    private String cachedMetadataLastModified;
+
+    public HTTPMetadataResolver(final HttpClient client, final String metadataURL) throws ResolverException {
+        this(null, client, metadataURL);
+    }
+
+    public HTTPMetadataResolver(final Timer backgroundTaskTimer, final HttpClient client, final String metadataURL)
+        throws ResolverException {
+        super(backgroundTaskTimer);
+
+        if (client == null) {
+            throw new ResolverException("HTTP client may not be null");
+        }
+        httpClient = client;
+
+        try {
+            metadataURI = new URI(metadataURL);
+        } catch (final URISyntaxException e) {
+            throw new ResolverException("Illegal URL syntax", e);
+        }
+    }
+
+    public String getMetadataURI() {
+        return metadataURI.toASCIIString();
+    }
+
+    @Override
+    protected void doDestroy() {
+        if (httpClient instanceof AutoCloseable) {
+            try {
+                ((AutoCloseable) httpClient).close();
+            } catch (final Exception e) {
+                log.error("Error closing HTTP client", e);
+            }
+        }
+        httpClient = null;
+        metadataURI = null;
+        cachedMetadataETag = null;
+        cachedMetadataLastModified = null;
+
+        super.doDestroy();
+    }
+
+    @Override
+    protected String getMetadataIdentifier() {
+        return metadataURI.toString();
+    }
+
+    @Override
+    protected byte[] fetchMetadata() throws ResolverException {
+        final HttpGet httpGet = buildHttpGet();
+        final HttpClientContext context = HttpClientContext.create();
+
+        try {
+            log.debug("{} Attempting to fetch metadata document from '{}'", getLogPrefix(), metadataURI);
+            return httpClient.execute(httpGet, context, response1 -> {
```

Comment:
`response1` is misleading :-) lets call it differently

Reply:
Oh yea, it's leftover from refactoring from the deprecated method. Renamed to just "response"